src/DebugConfigurationProvider.ts

```diff
@@ -60,6 +60,7 @@ export class DebugConfigurationProvider implements vscode.DebugConfigurationProv
       cwd: '${workspaceFolder}',
       console: 'integratedTerminal',
       internalConsoleOptions: 'neverOpen',
+      disableOptimisticBPs: true,
```

Comment:
my gut feeling is we probably don't want to add this for **all** debug config... shouldn't we just add this to the CRA [snippet](https://github.com/jest-community/vscode-jest/blob/6718f847a90e6fcae1faf847db289e456190b4b4/package.json#L167) in project.json instead? 

@stephtr probably is most familiar with this...

Reply:
> add this to the CRA snippet in project.json 

This is not specific to CRA. I use it for non React projects. 

And as i pointed is the recommended setting by Microsoft to run Jest: https://github.com/Microsoft/vscode-recipes/tree/master/debugging-jest-tests#configure-launchjson-file-for-your-test-framework

Reply:
Finally, in microsoft/vscode#60187 has all the reasoning as this is necessary to run jest properly specially https://github.com/microsoft/vscode/issues/60187#issuecomment-428408469

Reply:
looks like I was able to debug typescript projects without this flag, but js project requiring transpiling does benefit from having this flag. Just to play devil's advocate, if there is no downside of enabling this flag, why not use it for all the other debug recipes?

Reply:
> looks like I was able to debug typescript projects without this flag

typescript is not affected

>  if there is no downside of enabling this flag, why not use it for all the other debug recipes?

Seems its not as performant as disabled because needs to wait to the sourcemap be loaded to set the breakpoint. Although i did not notice any slowdown

Anyway, another possibility is to add an option disabled by default. If you can point me the place where to add i would be glad

Reply:
>> if there is no downside of enabling this flag, why not use it for all the other debug recipes?
>
> Seems its not as performant as disabled because needs to wait to the sourcemap be loaded to set the breakpoint. Although i did not notice any slowdown

According to the VS Code issue it can be (probably mostly for larger projects), that the breakpoints aren't ready when the tests are being executed. In that case one has to run the tests a second time or add a `debugger` statement to the top of the file and wait for a bit. The `disableOptimisticBPs` fix seems to be just kind of temporarily ("It's not a permanent solution but should unblock this at least."), but as long as it is used in vscode-recipes, I would set it by default in vscode-jest.

Reply:
By the way, if you already update this file, could you update the link on line 53 to [facebook.github.io](https://facebook.github.io/create-react-app/docs/debugging-tests#debugging-tests-in-visual-studio-code)?

Reply:
> According to the VS Code issue it can be (probably mostly for larger projects)

I get this issue in small projects like https://github.com/blikblum/nextbone-state/blob/master/test/unit/resource.spec.js

> In that case one has to run the tests a second time or add a debugger statement to the top of the file and wait for a bit

I found this by myself. Before this option, I put an aleatory breakpoint at top of file and when is hit i put in the correct place. 
With this option i not needed to do it anymore.